packages/payments-service/src/payments/controller.ts

```diff
@@ -1,24 +1,28 @@
-import { Headers, Controller, Post, Body, UseGuards, Patch, Param } from '@nestjs/common'
+import { Headers, Controller, Get, Body, UseGuards, Patch, Param } from '@nestjs/common'
 import { SessionGuard } from '../session/session-guard'
-import { PaymentsDto, PaymentsHeaders } from './dto'
+import { PaymentWithPropertyModel } from '../types/payment'
+import { PaymentsDto, PaymentsHeaders, PaymentsParams } from './dto'
 import { PaymentsProvider } from './provider'
 
 @Controller('payments')
 @UseGuards(SessionGuard)
 export class PaymentsController {
   constructor(private readonly paymentsProvider: PaymentsProvider) {}
 
-  @Post('/:paymentId')
-  async getPayment(@Headers() paymentsHeaders: PaymentsHeaders, @Param() paymentId: string): Promise<void> {
-    this.paymentsProvider.getPayment(paymentsHeaders, paymentId)
+  @Get('/:paymentId')
+  async getPayment(
+    @Headers() paymentsHeaders: PaymentsHeaders,
+    @Param() { paymentId }: PaymentsParams,
+  ): Promise<PaymentWithPropertyModel> {
+    return await this.paymentsProvider.getPayment(paymentsHeaders, paymentId)
```

Comment:
```suggestion
    return this.paymentsProvider.getPayment(paymentsHeaders, paymentId)
```

Don't need the await here